relay-server/src/metrics_extraction/transactions/mod.rs

```diff
@@ -322,13 +323,15 @@ impl TransactionExtractor<'_> {
                     } else {
                         tags.clone()
                     },
-                    score_profile_version: if is_performance_score {
-                        event
-                            .tag_value("sentry.score_profile_version")
-                            .map(|version| version.to_string())
-                    } else {
-                        None
-                    },
+                    score_profile_version: is_performance_score
+                        .then(|| {
+                            event
+                                .context::<PerformanceScoreContext>()
+                                .unwrap()
+                                .score_profile_version
+                                .value()
+                        })
+                        .and_then(|version| version.map(|s| s.to_string())),
```

Comment:
If there's no `PerformanceScoreContext` this panics. Something like the following should work:
```suggestion
                    score_profile_version: is_performance_score
                        .then(|| {
                            event.context::<PerformanceScoreContext>()
                        })
                        .map(|context| context.score_profile_version.value())
                        .and_then(|version| version.map(|s| s.to_string())),
```

Reply:
@iker-barriocanal thanks for catching this!

Reply:
@iker-barriocanal really appreciate you catching that! I see now that the use of `.unwrap()` is dangerous. When I put in your code snippet I get this error:

```rust
error[E0609]: no field `score_profile_version` on type `std::option::Option<&relay_event_schema::protocol::PerformanceScoreContext>`
   --> relay-server/src/metrics_extraction/transactions/mod.rs:328:48
    |
328 |                         .map(|context| context.score_profile_version.value())
    |                                                ^^^^^^^^^^^^^^^^^^^^^ unknown field
    |
help: one of the expressions' fields has a field of the same name
    |
328 |                         .map(|context| context.unwrap().score_profile_version.value())
    |                                                +++++++++
```

In this case is it safe to use `unwrap` because it's in the context of `.and_then`?

Reply:
This code snippet seems to work, however I feel like the last line feels like it's unnecessary/could be improved. Thoughts on this?

```rust
                    score_profile_version: is_performance_score
                        .then(|| event.context::<PerformanceScoreContext>())
                        .and_then(|context| context.unwrap().score_profile_version.value())
                        .and_then(|version| Some(version).cloned()),
```

Reply:
```rs
score_profile_version: is_performance_score
    .then(|| event.context::<PerformanceScoreContext>())
    .and_then(|context| context.unwrap().score_profile_version.value())
    .map(|version| version.clone()),
```

Reply:
Untested without unwrap:

```rs
score_profile_version: is_performance_score
    .then(|| event.context::<PerformanceScoreContext>())
    .and_then(|context| context?.score_profile_version.value())
    .map(|version| version.clone()),
```

Reply:
Thanks @Dav1dde, one of the linting tools gave me a warning about using an explicit closure to clone the version, so I just moved the call into the `.and_then` argument. This looks like it works without warnings

```rust
score_profile_version: is_performance_score
    .then(|| event.context::<PerformanceScoreContext>())
    .and_then(|context| context?.score_profile_version.value().cloned()),
```